packages/Config/RectorConfig.php

```diff
@@ -209,6 +215,9 @@ static function (
                 }
             );
         }
+
+        // for cache invalidation in case of change
+        SimpleParameterProvider::addParameter(Option::REGISTERED_RECTOR_RULES, $rectorClass);
```

Comment:
@TomasVotruba add parameter without merge seems cause double register:

```
   'registered_rector_rules' => array (2)
   |  0 => 'Rector\TypeDeclaration\Rector\StmtsAwareInterface\DeclareStrictTypesRector'
   |  1 => 'Rector\TypeDeclaration\Rector\StmtsAwareInterface\DeclareStrictTypesRector'
```

Reply:
@TomasVotruba here test case when rule not changed https://github.com/rectorphp/rector-src/pull/4903/commits/3ab9ae9e42acedff7f9bb7fb46a713bfa7c437bc at PR:

- https://github.com/rectorphp/rector-src/pull/4903

Reply:
@TomasVotruba the test case seems working ok now at PR:

- https://github.com/rectorphp/rector-src/pull/4903